### PR TITLE
fix(replica-scheduling): power off replica node should not rebuild new replica on same node

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#1992

#### What this PR does / why we need it:

- Before creating a new Replica CR, verify the availability of suitable nodes with sufficient disk to accommodate the replica. This prevents creating Replicas that cannot be scheduled.
- During node disk candidate selection, exclude nodes where the InstanceManager is not in a running state. This prevents creating redundant Replica CRs on nodes that haven't fully recovered after a power down.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

**Comment by c3y1huang** [comment]
Testing WIP.

**Comment by c3y1huang** [comment]
Test result: https://ci.longhorn.io/job/private/job/longhorn-e2e-test/669/

File: controller/volume_controller_test.go [hunk 1]
```diff
@@ -119,15 +119,38 @@ func (s *TestSuite) TestVolumeLifeCycle(c *C) {
...
 		r.Spec.DiskID = ""
 		r.Spec.DiskPath = ""
 		r.Spec.DataDirectoryName = ""
 	}
+	tc.expectVolume.Status.State = longhorn.VolumeStateCreating
+	tc.expectVolume.Status.CurrentImage = tc.volume.Spec.Image
+	tc.expectVolume.Status.Robustness = longhorn.VolumeRobustnessFaulted
+	testCases["volume create - replica creation failure"] = tc
+
+	// unable to create volume because no node to schedule
```
Comment (ejweber) [inline controller/volume_controller_test.go:137]: Can you help me to understand this test case? If I understand correctly, the one above (`volume create - replica creation failure`) is now expected. (We will try not to create replicas that cannot be scheduled.) In this test case, the replicas are already created but cannot be scheduled (as was the old behavior). Is this a state we expect to be in? Or is it just to cover a corner case that probably won't be triggered?

**Review by ejweber — APPROVED** [review]
LGTM.

I do have one question about a test case, but it is for my understanding and not a blocker.

**Review by innobead — APPROVED** [review]

**Comment by c3y1huang** [comment]
@mergify backport v1.5.x v1.6.x

File: controller/volume_controller_test.go (hunk 1 above)
Comment (c3y1huang) [reply controller/volume_controller_test.go:137]: This is likely a corner case. In my opinion, the unit test will give us a better understanding of the behavior.
